parsers/rhea/rhea.go

```diff
@@ -0,0 +1,430 @@
+	ContainsN    bool   `json:"containsn" db:"containsn"`
+	Minus        bool   `json:"minus" db:"minus"` // Only set to true if ContainsN == true to handle Nminus1
+	Plus         bool   `json:"plus" db:"plus"`   // Only set to true if ContainsN == true to handle Nplus1
+	Accession    string `json:"reactionParticipant" db:"ReactionParticipant"`
+	Compound     string `json:"compound" db:"compound"`
+}
+
+// Reaction represents a Rhea reaction. Substrates, Products, and SubstrateOrProducts are all ReactionSide accession
+// numbers, which can be linked to the ReactionParticipant's ReactionSide accession
+type Reaction struct {
+	ID                   int      `json:"id" db:"id"`
+	Directional          bool     `json:"directional" db:"directional"`
+	Accession            string   `json:"accession" db:"accession"`
+	Status               string   `json:"status" db:"status"`
+	Comment              string   `json:"comment" db:"comment"`
+	Equation             string   `json:"equation" db:"equation"`
+	HTMLEquation         string   `json:"htmlequation" db:"htmlequation"`
+	IsChemicallyBalanced bool     `json:"ischemicallybalanced" db:"ischemicallybalanced"`
+	IsTransport          bool     `json:"istransport" db:"istransport"`
+	Ec                   string   `json:"ec" db:"ec"`
+	Location             string   `json:"location" db:"location"`
+	Citations            []string `json:"citations"`
+	Substrates           []string `json:"substrates"`
+	Products             []string `json:"products"`
+	SubstrateOrProducts  []string `json:"substrateOrProducts"`
+}
+
+/******************************************************************************
+
+Parse functions
+
+These functions take in the rhea.rdf.gz dump file and return a Rhea struct,
+which contains all of the higher level structs
+
+******************************************************************************/
+
+// NewReaction returns a Reaction.
+func NewReaction(description Description, subclass Subclass) Reaction {
+	return Reaction{
+		ID:                   description.ID,
+		Directional:          subclass.Resource == "http://rdf.rhea-db.org/DirectionalReaction",
+		Accession:            description.Accession,
+		Status:               description.Status.Resource,
+		Comment:              description.Comment,
+		Equation:             description.Equation,
+		HTMLEquation:         description.HTMLEquation,
+		IsChemicallyBalanced: description.IsChemicallyBalanced,
+		IsTransport:          description.IsTransport,
+		Ec:                   description.EC.Resource,
+		Citations:            description.CitationStrings(),
+		Substrates:           description.SubstrateAccessionIDs(),
+		Products:             description.ProductAccessionIDs(),
+		SubstrateOrProducts:  description.SubstrateOrProductAccessionIDs(),
+		Location:             description.Location.Resource}
+}
+
+// NewCompound returns a Compound.
+func NewCompound(description Description, subclass Subclass) Compound {
```

Comment:
This should be refactored into `newSmallMolecule`, `newPolymer`... until there are no more switch statements in side any of these functions.

All the switch functions should occur in the main `parseRhea` function.